src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,49 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    static const QRegularExpression episodeRegex(
+                    "(?:^|[^A-Za-z0-9])(?:"
+
+                    //Format 1: s01e01
+                    "(?:s(\\d{1,})e(\\d{1,}))|"
+
+                    //Format 2: 01x01
+                    "(?:(\\d{1,})x(\\d{1,}))|"
+
+                    //Format 3: 2017.01.01
+                    "(\\d{4}\\.\\d{2}\\.\\d{2})"
+
+                    ")(?:[^A-Za-z0-9]|$)",
```

Comment:
As much as I don't want to suggest delaying this PR as I very much want it to be merged, I have to agree with @thalieht about format 3. There is no standard at all with torrents, so I'd expect at some point this would fail to match something in someone's feed.

A better regex is probably: `(\d{4}[_.\-x]\d{1,2}[_.\-x]\d{1,2}|\d{1,2}[_.\-x]\d{1,2}[_.\-x]\d{4})`
or `(\\d{4}[_.\\-x]\\d{1,2}[_.\\-x]\\d{1,2}|\\d{1,2}[_.\\-x]\\d{1,2}[_.\\-x]\\d{4})` if it's escaped properly.

That will match things like `2014.05.03`, `5.3.2014`, `2014_05_03`, `3-5-2014`, `2014x05x03` and so on. I think that's a good balance between being lenient enough to be unlikely to miss something and not so lenient as to match someone erroneously.